File: algorithms/1_easy/M/01758_minimum_changes_to_make_alternating_binary_string/solution.c
int minOperations(const char *restrict s) {
    size_t zero_first = 0;
    size_t one_first = 0;

    for (size_t i = 0; s[i] != '\0'; i += 1) {
        if ((i & 1) == 0) {
            switch (s[i]) {
                case '0': 
                    one_first += 1;
                    break;
                case '1':
                    zero_first += 1;
                    break;
                default:
                    return -1;
            }
        } else {
            switch (s[i]) {
                case '0': 
                    zero_first += 1;
                    break;
                case '1':
                    one_first += 1;
                    break;
                default:
                    return -1;
            }
        }
    }

    return (int)MIN(zero_first, one_first);
}

File: algorithms/1_easy/M/01758_minimum_changes_to_make_alternating_binary_string/solution.rs
impl Solution {
    pub fn min_operations(s: String) -> i32 {
        let (mut zero_first, mut one_first) = (0_usize, 0_usize);

        for (i, &byte) in s.as_bytes().into_iter().enumerate() {
            if (i & 1) == 0 {
                match byte {
                    b'0' => one_first += 1,
                    b'1' => zero_first += 1,
                    _ => unreachable!(),
                }
            } else {
                match byte {
                    b'0' => zero_first += 1,
                    b'1' => one_first += 1,
                    _ => unreachable!(),
                }
            }
        }

        zero_first.min(one_first).try_into().unwrap()
    }
}
